File: src/test/java/com/everyonewaiter/domain/notification/DiscordEmbedsTest.java
package com.everyonewaiter.domain.notification;

import static com.everyonewaiter.domain.notification.NotificationFixture.createDiscordEmbed;
import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;

import java.util.ArrayList;
import java.util.List;
import org.junit.jupiter.api.Test;

class DiscordEmbedsTest {

  @Test
  void constructor() {
    List<DiscordEmbed> embeds = new ArrayList<>();
    embeds.add(createDiscordEmbed());
    embeds.add(createDiscordEmbed());

    DiscordEmbeds embeds1 = new DiscordEmbeds(embeds);
    DiscordEmbeds embeds2 = new DiscordEmbeds(createDiscordEmbed());

    assertThat(embeds1.embeds()).hasSize(2);
    assertThat(embeds2.embeds()).hasSize(1);
  }

  @Test
  void constructorFail() {
    List<DiscordEmbed> embeds = new ArrayList<>();
    for (int i = 0; i < 11; i++) {
      embeds.add(createDiscordEmbed());
    }

    assertThatThrownBy(() -> new DiscordEmbeds(List.of()))
        .isInstanceOf(IllegalArgumentException.class);
    assertThatThrownBy(() -> new DiscordEmbeds(embeds))
        .isInstanceOf(IllegalArgumentException.class);
  }

}

File: src/test/java/com/everyonewaiter/domain/notification/DiscordEmbedsTest.kt
package com.everyonewaiter.domain.notification

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

class DiscordEmbedsTest {

  @Test
  fun `디스코드 임베드 생성`() {
    val embeds1 = DiscordEmbeds(createDiscordEmbed(), createDiscordEmbed())

    assertThat(embeds1.embeds).hasSize(2)
  }

  @Test
  fun `디스코드 임베드의 개수가 0이거나 10을 초과하면 생성 실패`() {
    assertThatThrownBy {
      DiscordEmbeds(emptyList())
    }.isInstanceOf(IllegalArgumentException::class.java)

    assertThatThrownBy {
      DiscordEmbeds(List(11) { createDiscordEmbed() })
    }.isInstanceOf(IllegalArgumentException::class.java)
  }
}
